aomi/cli.py

```diff
@@ -137,33 +155,36 @@ def action_runner(operation):
         sys.exit(0)
 
     client = aomi.vault.client(operation, opt)
-    if operation == 'extract_file':
-        if len(args) == 3:
-            aomi.render.raw_file(client, args[1], args[2], opt)
-            sys.exit(0)
-    elif operation == 'environment':
-        if len(args) >= 2:
-            paths = args[1:]
-            aomi.render.env(client, paths, opt)
-            sys.exit(0)
-    elif operation == 'aws_environment':
-        if len(args) == 2:
-            aomi.render.aws(client, args[1], opt)
-            sys.exit(0)
-    elif operation == 'seed':
-        if len(args) == 1:
-            aomi.validation.gitignore(opt)
-            aomi.vault.seed(client, opt)
-            sys.exit(0)
-    elif operation == 'template':
-        if len(args) >= 4:
-            paths = args[3:]
-            aomi.render.template(client, args[1], args[2], paths, opt)
-            sys.exit(0)
-    elif operation == 'token':
-        if len(args) == 1:
-            print(client.token)
-            sys.exit(0)
+    if operation == 'extract_file' and len(args) == 3:
```

Comment:
Maybe change this giant if statement into a dictionary of operations, and break the sys.exit(0) out of each case.
Possibly a similar idea with the validators for each operation. For example
```python
operation_dict = {
    ('environment', 2) : do_environment()
    ('environment', 3) : do_environment_with_extra_opt()
    ('seed',1) : do_seed()
}

validator_dict = {

   ('environment',2): env_validate(2),
   ('environment',3): env_validate(3)
   ('seed', 1) :seed_validate
} 

def seed_validate(context):
   len(context.args) == 1

def env_validate(args_num):
   lambda context:
     len(context.args) == args

def environment(context):
   """does work"""
   ...

def seed(context)
    """ does seed"
    ....

def run(args):
    """Your parse args"""
    parsed = parse_args(args)
    op = operations[(parsed.operation, len(parse))]
    validate = validators[(parsed.operation, len(parse))]
    context  = {
       'args': args,
       'parsed': parsed,
       'client': client,
       'validator' : validate
}
     (valid_result,message)  = validate(context)
     if(valid_result):
         op(context)
         exit(0)
     else
         print message
         exit(1)
```

Or maybe not... I got caught up.